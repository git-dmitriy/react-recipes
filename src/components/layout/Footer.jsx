import GithubIcon from '../icons/GithubIcon';
import { FaGithub } from 'react-icons/fa';

export function Footer() {
  return (
    <footer className='bg-gray-900 py-3 text-white'>
      <div className='container mx-auto'>
        <div className='flex justify-center items-center'>
          Â© {new Date().getFullYear()} Dmitriy Shalberkin
          <a
            className='mx-4 w-10 text-4xl hover:text-gray-400 transition duration-300'
            target='_blank'
            rel='noreferrer'
            href='https://github.com/git-dmitriy'
          >
            <GithubIcon />
          </a>
        </div>
      </div>
    </footer>
  );
}
