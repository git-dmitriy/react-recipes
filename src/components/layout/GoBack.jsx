import { useEffect, useState } from 'react';
import { useNavigate, useLocation } from 'react-router-dom';
import { FaArrowLeft } from 'react-icons/fa';

const GoBack = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const [isHome, setIsHome] = useState(false);

  useEffect(() => {
    if (location.pathname === '/') {
      setIsHome(true);
    } else {
      setIsHome(false);
    }
  }, [location]);

  return (
    <div className='flex items-center sm:mx-6'>
      {!isHome ? (
        <button className='w-5 text-3xl' onClick={() => navigate(-1)}>
          <FaArrowLeft className='fill-current' />
        </button>
      ) : (
        <div className='w-5 text-transparent'>
          <FaArrowLeft className='fill-current text-transparent' />
        </div>
      )}
    </div>
  );
};

export default GoBack;
