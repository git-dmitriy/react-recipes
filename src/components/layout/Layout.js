export const Layout = ({ children }) => {
  return (
    <main className='text-gray-600'>
      <div className='container px-5 pt-2 pb-24 mx-auto flex flex-col'>
        <div className='w-full lg:w-6/6 2xl:w-4/6 mx-auto '>{children}</div>
      </div>
    </main>
  );
};
