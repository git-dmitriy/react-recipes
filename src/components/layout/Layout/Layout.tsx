export const Layout: React.FC = ({ children }) => {
  return (
    <div className='container px-3 sm:px-5 pt-32 pb-24 mx-auto flex flex-col'>
      <div className='w-full lg:w-6/6 2xl:w-5/6 mx-auto relative'>
        {children}
      </div>
    </div>
  );
};
