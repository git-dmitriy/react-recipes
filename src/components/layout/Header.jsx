import { useNavigate, NavLink } from 'react-router-dom';
import { Search } from '../Search/Search';
import { GoBack } from './GoBack';
import { BsFillBookmarkFill } from 'react-icons/bs';

export const Header = () => {
  const { push } = useNavigate();

  const handleSearch = (searchQuery) => {
    if (searchQuery.trim().length !== 0) {
      push({
        pathname: '/recipes',
        search: `search=${searchQuery}`,
      });
    }
  };

  const textLinkClassName =
    'border-b-2 border-transparent font-bold outline-none hover:border-red-500 mx-1.5 sm:mx-6 focus:outline-none focus:border-red-500 transition';
  const iconLinkClassName =
    'border-b-2 border-transparent outline-none hover:text-red-500 mx-1.5 sm:mx-6 focus:outline-none focus:text-red-500 transition text-2xl';
  const activeTextLink = ' border-b-2 border-red-500';
  const activeIconLink = ' text-red-500';

  return (
    <nav className='bg-yellow-300 shadow dark:bg-gray-800 fixed top-0 w-full backdrop-blur-md backdrop-filter bg-opacity-50'>
      <ul className='container flex items-center text-lg justify-around sm:justify-center px-2 sm:px-6 mx-auto capitalize dark:text-gray-300'>
        <li>
          <GoBack />
        </li>
        <li>
          <Search cb={handleSearch} />
        </li>

        <li>
          <NavLink
            end
            to='/'
            className={({ isActive }) =>
              textLinkClassName + (isActive ? activeTextLink : '')
            }
          >
            Recipes
          </NavLink>
        </li>

        <li>
          <NavLink
            to='favorites'
            className={({ isActive }) =>
              iconLinkClassName + (isActive ? activeIconLink : '')
            }
          >
            <BsFillBookmarkFill className='fill-current' />
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
