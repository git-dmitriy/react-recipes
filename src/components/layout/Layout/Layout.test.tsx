import renderer from 'react-test-renderer';
import { Layout } from 'components/layout/Layout';

it('should render correctly', () => {
  const tree = renderer
    .create(
      <Layout>
        <div />
      </Layout>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
