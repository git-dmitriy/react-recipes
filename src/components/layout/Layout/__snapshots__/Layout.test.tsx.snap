// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container px-3 sm:px-5 pt-32 pb-24 mx-auto flex flex-col"
>
  <div
    className="w-full lg:w-6/6 2xl:w-5/6 mx-auto relative"
  >
    <div />
  </div>
</div>
`;
